### DISCUSSION TITLE
DateInput edit the text field, then close on pressing Enter

### DISCUSSION THREAD

**Discussion by andreidiaconescu** [mantinedev/mantine#7026] [Q&A]
Hello,

I would need to be able to edit the date text field, and then when i press Enter the calendar popup should close.
I managed to implement all, except the closing: how can i "programatically" like, close the calendar popup.
Any hint ?

thank you.

**Comment by Kenzo-Wada** [comment]
In that case, because DateInput's default behaviour is opening Datepicker by trigger of onClick, I think you have to make custom component using Input and Datepicker.

**Comment by andreidiaconescu** [comment]
thank you @Kenzo-Wada 
i guess, in my case that would not be a small amount of work ...
i guess it would be great though, if an extra prop was added to allow "toggeling" the calendar when clicking on the input
